src/snowflake/connector/cursor.py

```diff
@@ -929,8 +929,8 @@ def execute(
 
         if _do_reset:
             self.reset()
-        command = command.strip(" \t\n\r") if command else None
-        if not command:
+        command = command.strip(" \t\n\r") if command else ""
+        if not command and not _dataframe_ast:
```

Comment:
Can we maybe clarify the logging logic here? If I understand this correctly, basically in the case of _dataframe_ast =True, command will be "".  Should we log out the base64 string in debug mode? I.e.,

```
if not command:
   if _dataframe_ast:
         logger.debug("query[AST base64]: [%s]", _dataframe_ast)
   else:
         logger.warning("execute: no query is given to execute")
         return      
```

Reply:
Good Idea, let me change it.

Reply:
changed